tpm2/tpm2_test.go

```diff
@@ -1363,3 +1363,47 @@ func TestCreatePrimaryRawTemplate(t *testing.T) {
 		t.Errorf("got key exponent %v, want %v", pubRSA.E, defaultKeyParams.RSAParameters.Exponent)
 	}
 }
+
+func TestMatchesTemplate(t *testing.T) {
```

Comment:
Add tests for other types of template (ECC and SymCipher)
Ideally use subtests here (`t.Run`)